setup.py

```diff
@@ -141,9 +141,20 @@ def _set_builtin(name, value):
     packages=find_packages(),
     package_data={'cf_units': list(file_walk_relative('cf_units/etc',
                                                       remove='cf_units/'))},
-    install_requires=load('requirements.txt'),
+    install_requires=['antlr4-python3-runtime==4.7.2',
+                      'cftime>=1.2',
+                      'numpy',
+                      # udunits2 cannot be installed with pip, and it is
+                      #  expected to be installed separately.
+                      ],
     setup_requires=['pytest-runner', 'numpy', 'cython'],
-    tests_require=load('requirements-dev.txt'),
+    tests_require=['codecov',
+                   'cython',
```

Comment:
The tests shouldn't explicitly need `cython`, as `pip install -e .` with the associated `pyproject.toml` will build the cython extensions and install `cf-units` in developer mode, and the tests will just run thanks to PEP517/518 :+1: 

I've tested this pattern with [SciTools-incubator/python-stratify](https://github.com/SciTools-incubator/python-stratify), which like `cf-units`, also used `cython` to build extensions - so you could see the pattern we've adopted there for fuller clarity.

Reply:
I'm well aware there's a lot more to do on architecture, but I've been trying to keep this PR as the minimum necessary to migrate the CI. Do you consider this one a deal breaker that can't wait?

Reply:
Cool. Totally understood. It can certainly wait, but it would be good to revisit this before we push another release :+1: